src/System.Private.CoreLib/shared/System/AppDomain.cs

```diff
@@ -184,26 +184,21 @@ public static void Unload(AppDomain domain)
 
         public static bool MonitoringIsEnabled
         {
-            get { return false; }
+            get { return true; }
             set
             {
                 if (!value)
                 {
                     throw new ArgumentException(SR.Arg_MustBeTrue);
                 }
-                throw new PlatformNotSupportedException(SR.PlatformNotSupported_AppDomain_ResMon);
             }
         }
 
-        public long MonitoringSurvivedMemorySize { get { throw CreateResMonNotAvailException(); } }
+        public long MonitoringSurvivedMemorySize => MonitoringSurvivedProcessMemorySize;
 
-        public static long MonitoringSurvivedProcessMemorySize { get { throw CreateResMonNotAvailException(); } }
+        public static long MonitoringSurvivedProcessMemorySize => GC.GetGCMemoryInfo().HeapSizeBytes;
 
-        public long MonitoringTotalAllocatedMemorySize { get { throw CreateResMonNotAvailException(); } }
-
-        public TimeSpan MonitoringTotalProcessorTime { get { throw CreateResMonNotAvailException(); } }
-
-        private static Exception CreateResMonNotAvailException() => new InvalidOperationException(SR.PlatformNotSupported_AppDomain_ResMon);
+        public long MonitoringTotalAllocatedMemorySize => GC.GetTotalAllocatedBytes();
```

Comment:
>MonitoringTotalAllocatedMemorySize [](start = 20, length = 34)

I'm wondering if it makes sense for this to also give users an option for precious monitoring (which would call GC.GetTotalAllocatedBytes with true). I don't have a strong opinion on this - perhaps someone who might want to use this should comment - @adamsitnik?

Reply:
> I'm wondering if it makes sense for this to also give users an option for precious monitoring (which would call GC.GetTotalAllocatedBytes with true)

@Maoni0, which of `GetTotalAllocatedBytes(true)` or `GetTotalAllocatedBytes(false)` maps better to `MonitoringTotalAllocatedMemorySize` on netfx today?  We can just use whichever is closer, since the whole point of this is to enable existing such code to keep working.  Someone who wants the other semantics can just use `GetTotalAllocatedBytes` directly, or other such APIs we expose.

Reply:
FWIW As the end user of this API I would expect it to call `GC.GetTotalAllocatedBytes(true)`. The way I see it is that `MonitoringTotalAllocatedMemorySize` requires `AppDomain.MonitoringIsEnabled = true;` on Full .NET Framework which means that the monitoring has a cost.

Reply:
> Full .NET Framework which means that the monitoring has a cost.

This cost was in per-AppDomain accounting, e.g. enabling monitoring made the AppDomain switching more expensive.

These APIs have not provided accurate results in .NET Framework. `GC.GetTotalAllocatedBytes(/*false*/)` is a good approximation for what these APIs have done in .NET Framework.